### PR TITLE
Add charts for GPP-app and GPP publicatiebank

### PR DESCRIPTION
**Pull request by SilviaAmAm**
Fixes https://github.com/GPP-Woo/GPP-publicatiebank/issues/107

### CODE REVIEW

**Comment by sergei-maertens** [comment]
> Todo:
> 
> * https://github.com/helm/chart-releaser-action?tab=readme-ov-file#pre-requisites => Check if in Settings/Pages the Source Branch is `gh-pages` (I don't have the permissions to do it)

@SilviaAmAm I can confirm this is how it's set up :)

File: charts/GPP-publicatiebank/Chart.yaml [hunk 1]
```diff
@@ -0,0 +1,17 @@
+apiVersion: v2
+name: gpp-publicatiebank
+description: A helm chart for the ICATT Openbaar-Documenten-Publicatie-Component
```
Comment (sergei-maertens) [inline charts/GPP-publicatiebank/Chart.yaml:3]:
```suggestion
description: A helm chart for the ICATT GPP-Publicatiebank
```
we've settled on the component names now ;)

File: charts/GPP-publicatiebank/README.md [hunk 2]
```diff
@@ -0,0 +1,101 @@
+# gpp-publicatiebank
+
+![Version: 0.1.0](https://img.shields.io/badge/Version-0.1.0-informational?style=flat-square) ![Type: application](https://img.shields.io/badge/Type-application-informational?style=flat-square) ![AppVersion: 1.16.0](https://img.shields.io/badge/AppVersion-1.16.0-informational?style=flat-square)
```
Comment (sergei-maertens) [inline charts/GPP-publicatiebank/README.md:3]: where does this AppVersion come from?

File: charts/GPP-publicatiebank/templates/configmap.yaml [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+  {{- end }}
+  {{- if .Values.settings.oidc.roleClaimType }}
+  OIDC_ROLE_CLAIM_TYPE: {{ .Values.settings.oidc.roleClaimType | toString | quote }}
+  {{- end }}
+  {{- if .Values.settings.oidc.idClaimType }}
+  OIDC_ID_CLAIM_TYPE: {{ .Values.settings.oidc.idClaimType | toString | quote }}
+  {{- end }}
+  {{- if .Values.settings.oidc.nameClaimType }}
+  OIDC_NAME_CLAIM_TYPE: {{ .Values.settings.oidc.nameClaimType | toString | quote }}
+  {{- end }}
```
Comment (sergei-maertens) [inline charts/GPP-publicatiebank/templates/configmap.yaml:31]: I don't recognize these envvars :thinking:

**Review by sergei-maertens — COMMENTED** [review]
I think you may be confusing some component names?

* gpp-publicatiebank: django backend (formerly ODRC/woo-publications)
* gpp-app: .net app (formerly ODPC)
* gpp-burgerportaal: .net app (I think), formerly ODBP

A little more documentation is available here: https://gpp-publicatiebank.readthedocs.io/en/latest/introduction/index.html

**Comment by SilviaAmAm** [comment]
> I think you may be confusing some component names?

I think so, so if I understand correctly: 

- gpp-publicatiebank: old registratie component
- gpp-app: old icatt-odpc 

I will swap them

File: charts/GPP-publicatiebank/README.md (hunk 2 above)
Comment (SilviaAmAm) [reply charts/GPP-publicatiebank/README.md:3]: They come from the Chart.yaml (which we have added manually). I think it was the latest stable tag when we deployed it last :thinking:

File: charts/GPP-publicatiebank/templates/configmap.yaml (hunk 3 above)
Comment (SilviaAmAm) [reply charts/GPP-publicatiebank/templates/configmap.yaml:31]: I took them from here: https://github.com/GPP-Woo/GPP-app/pkgs/container/gpp-app#omgevingsvariabelen

File: charts/GPP-publicatiebank/README.md [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+| settings.environmentLabelName | string | `""` | Environment information to display, defaults to the value of ENVIRONMENT. Only displayed when SHOW_ENVIRONMENT is set to True. You can set this to strings like OpenGem PROD or simply PROD, depending on your needs. |
+| settings.environment_background_color | string | `""` |  |
+| settings.environment_label | string | `""` | Environment information to display, defaults to the value of ENVIRONMENT. Only displayed when SHOW_ENVIRONMENT is set to True. You can set this to strings like OpenGem PROD or simply PROD, depending on your needs. |
+| settings.isHttps | bool | `true` |  |
+| settings.secretKey | string | `""` | Generate secret key at https://djecrety.ir/ |
+| settings.sentry.dsn | string | `""` |  |
+| settings.showLabelEnvironment | bool | `false` | Display environment information in the header in the admin. Defaults to True. Environment information is only displayed to logged in users. |
+| settings.show_label_environment | string | `"false"` | Display environment information in the header in the admin. Defaults to True. Environment information is only displayed to logged in users. |
```
Comment (sergei-maertens) [inline charts/GPP-publicatiebank/README.md:135]: same setting but different casing?

File: charts/GPP-publicatiebank/templates/configmap.yaml [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+  DJANGO_SETTINGS_MODULE: {{ .Values.settings.djangoSettingsModule | toString | quote }}
+  {{- if .Values.settings.useXForwardedHost }}
+  USE_X_FORWARDED_HOST: "True"
+  {{- end }}
+  ENVIRONMENT: {{ .Values.settings.environment | default (include "gpp-publicatiebank.fullname" .) }}
+  ALLOWED_HOSTS: "{{ include "gpp-publicatiebank.fullname" . }},{{ include "gpp-publicatiebank.fullname" . }}.{{ .Release.Namespace }},{{ .Values.settings.allowedHosts | replace " " "" | toString }}"
+  {{- if .Values.tags.redis }}
+  CACHE_DEFAULT: {{ printf "%s-master.%s:6379/0" (include "common.names.fullname" .Subcharts.redis) .Release.Namespace | toString | quote }}
+  CACHE_AXES: {{ printf "%s-master.%s:6379/0" (include "common.names.fullname" .Subcharts.redis) .Release.Namespace | toString | quote }}
+  CACHE_OIDC: {{ printf "%s-master.%s:6379/0" (include "common.names.fullname" .Subcharts.redis) .Release.Namespace | toString | quote }}
```
Comment (sergei-maertens) [inline charts/GPP-publicatiebank/templates/configmap.yaml:17]: unused envvar

**Review by sergei-maertens — APPROVED** [review]
I've looked at the Django project, can't say much about the gpp-app :)

File: charts/GPP-app/Chart.yaml [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+name: gpp-app
+description: A helm chart for the ICATT GPP app.
+
+type: application
+
+# Chart version 
+version: 0.1.0
+appVersion: 0.0.0
+
+dependencies:
```
Comment (felixcicatt) [inline charts/GPP-app/Chart.yaml:11]: GPP-app doesn't need a dependency on Redis

File: charts/GPP-app/templates/deployment.yaml [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+                name: {{ .Values.existingSecret | default (include "gpp-app.fullname" .) }}
+            - configMapRef:
+                name: {{ include "gpp-app.fullname" . }}
+          ports:
+            - name: http
+              containerPort: 8080
+              protocol: TCP
+          livenessProbe:
+            httpGet:
+              path: /
```
Comment (felixcicatt) [inline charts/GPP-app/templates/deployment.yaml:54]:
```suggestion
              path: /healthz
```

File: charts/GPP-app/templates/deployment.yaml [hunk 8]
```diff
@@ -0,0 +1,81 @@
...
+              port: http
+              httpHeaders:
+                - name: Host
+                  value: {{ include "gpp-app.fullname" . | quote }}
+            {{- toYaml .Values.livenessProbe | nindent 12 }}
+          readinessProbe:
+            httpGet:
+              path: /
```
Comment (felixcicatt) [inline charts/GPP-app/templates/deployment.yaml:62]:
```suggestion
              path: /healthz
```

**Review by felixcicatt — CHANGES_REQUESTED** [review]
If we can remove the dependency on Redis in the GPP-app, everything looks good to me, thanks!

File: charts/GPP-publicatiebank/README.md (hunk 4 above)
Comment (SilviaAmAm) [reply charts/GPP-publicatiebank/README.md:135]: The snake case version can be removed!

File: charts/GPP-publicatiebank/templates/configmap.yaml (hunk 5 above)
Comment (SilviaAmAm) [reply charts/GPP-publicatiebank/templates/configmap.yaml:17]: Removed `CACHE_OIDC`

**Pull request merged**